modules/yardstick/src/main/java/org/apache/ignite/yardstick/cache/IgniteTxSuspendResumeBenchmark.java

```diff
@@ -0,0 +1,113 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.yardstick.cache;
+
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteTransactions;
+import org.apache.ignite.transactions.Transaction;
+import org.apache.ignite.transactions.TransactionState;
+import org.apache.ignite.yardstick.cache.model.SampleValue;
+import org.yardstickframework.BenchmarkConfiguration;
+
+/**
+ * Ignite benchmark that performs suspend\resume operations.
+ * Concurrent threads must operate different keys, so there will be no waiting on key locks.
+ * So, all keys are separated on different groups, defined by first key and last key(first key plus keys number per thread).
+ * Groups have no intersections.
+ */
+public class IgniteTxSuspendResumeBenchmark extends IgniteCacheAbstractBenchmark<Integer, Object> {
+    /** Transactions. */
+    private IgniteTransactions txs;
+
+    /** ExecutorService. */
+    private ExecutorService ex;
+
+    /**
+     * First keyCounter that can be used by transaction in certain thread.
+     * The last one can be calculated based on keys per thread
+     */
+    private ThreadLocal<Integer> firstKey = new ThreadLocal<Integer>() {
+        /** Helps to calculate first key for every thread. */
+        private AtomicInteger keyCntr = new AtomicInteger(0);
+
+        @Override protected Integer initialValue() {
+            return keyCntr.getAndAdd(keysNumbPerThread);
+        }
+    };
+
+    /** Number of keys, allocated to every thread. */
+    private int keysNumbPerThread = 0;
+
+    /** {@inheritDoc} */
+    @Override public void setUp(BenchmarkConfiguration cfg) throws Exception {
+        super.setUp(cfg);
+
+        txs = ignite().transactions();
+
+        ex = Executors.newFixedThreadPool(cfg.threads());
+
+        keysNumbPerThread = (args.range() > cfg.threads() ? args.range() / cfg.threads() : args.range());
+    }
+
+    /** {@inheritDoc} */
+    @Override public void tearDown() throws Exception {
+        super.tearDown();
+
+        ex.shutdown();
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean test(Map<Object, Object> ctx) throws Exception {
```

Comment:
`Transaction` creation and `key` putting are parts of your benchmark. So the result will not pure. I think it's possible to move write to javadoc how to calculate a pure result. Like `score_of_this_benchmark - score_of_IgnitePutTxBenchmark`.